src/test/ui/proc-macro/auxiliary/api/parse.rs

```diff
@@ -0,0 +1,20 @@
+use proc_macro::Literal;
+
+pub fn test() {
+    test_parse_literal();
+}
+
+fn test_parse_literal() {
+    assert_eq!("1".parse::<Literal>().unwrap().to_string(), "1");
+    assert_eq!("1.0".parse::<Literal>().unwrap().to_string(), "1.0");
+    assert_eq!("'a'".parse::<Literal>().unwrap().to_string(), "'a'");
+    assert_eq!("\"\n\"".parse::<Literal>().unwrap().to_string(), "\"\n\"");
+
```

Comment:
Might be good to test prefixed/suffixed literals too (`b""`, `10u16`), since the parsing code for those is about to change due to the edition changes.

Reply:
Added tests of `b""`, `r##""##`, and `10ulong` which all naively appear like they could be interpreted as multiple tokens but are in fact single literals.